test/unit/nox/_shared_test.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+
+import noxconfig
+from exasol.toolbox.nox._shared import (
+    DEFAULT_PATH_FILTERS,
+    python_files,
+)
+
+
+@pytest.fixture(scope="session")
+def package_directory():
+    return "toolbox"
+
+
+@pytest.fixture(scope="session")
+def tmp_directory(tmp_path_factory):
+    return tmp_path_factory.mktemp("data")
+
+
+@pytest.fixture(scope="session")
+def path_filter_directory():
+    return "path_filter"
+
+
+@pytest.fixture(scope="session")
+def directories(package_directory, path_filter_directory):
+    yield DEFAULT_PATH_FILTERS.union({package_directory, path_filter_directory})
+
+
+@pytest.fixture(scope="session")
+def create_files(tmp_directory, directories):
+    file_list = []
+    for directory in directories:
+        directory_path = tmp_directory / directory
+        directory_path.mkdir(parents=True, exist_ok=True)
+
+        file_path = directory_path / f"{directory}-dummy.py"
+        file_path.touch()
+        file_list.append(file_path)
+
+    yield file_list
+
+
+def test_python_files(
+    monkeypatch, tmp_directory, create_files, package_directory, path_filter_directory
+):
+    object.__setattr__(
```

Comment:
Could you please explain this? 
* Where does `object` come from? 
* Would it be possible to simply use `noxconfig.PROJECT_CONFIG.path_filters=(path_filter_directory,)`?

Reply:
OK I think I got it.

Maybe the following comment would be helpful?

Use builtin [object](https://docs.python.org/3/library/functions.html#object) to modify attribute `path_filters`
of frozen dataclass instance.